tests/unit/src/kakarot/accounts/contract/test_contract_account.py

```diff
@@ -12,11 +12,13 @@
 class TestContractAccount:
     @pytest.mark.parametrize("bytecode_len", [0, 15, 16, 17, 30, 31, 32, 33])
     async def test_should_store_code(
-        self, contract_account: StarknetContract, bytecode_len
+        self, contract_account: StarknetContract, bytecode_len, kakarot
     ):
         bytecode = [random.randint(0, 255) for _ in range(bytecode_len)]
 
         with traceit.context("contract_account"):
-            await contract_account.write_bytecode(bytecode).execute(caller_address=1)
+            await contract_account.write_bytecode(bytecode).execute(
+                caller_address=kakarot.contract_address
+            )
```

Comment:
can you add a test class for `initialize` like

```
class TestContractAccount:
    class TestInitialize:
        def test_should_run_only_once():
            await contract_account.initialize()
            with kakarot_error():
                await contract_account.initialize()

        def test_should_set_ownership()
        def test_should_give_inifinite_allowance_to_kakarot()
```

Reply:
Implemented :)
Feel free to check if it mirrors what you where suggesting.